django_declarative_apis/machinery/filtering.py

```diff
@@ -67,6 +68,34 @@ def _get_unexpanded_field_value(inst, field_name, field_type):
         return {display_key: getattr(obj, display_key)}
 
 
+def _is_relation(field):
+    return field.is_relation or getattr(field, "is_fake_relation", False)
+
+
+def _is_reverse_relation(field):
+    return isinstance(field, ForeignObjectRel)
+
+
+def _cache_related_instance(inst, field_name, model_cache):
+    try:
+        field_meta = inst._meta.get_field(field_name)
+    except (AttributeError, FieldDoesNotExist):
+        return
+    if not _is_relation(field_meta) or _is_reverse_relation(field_meta):
+        return  # no `attname` in reverse relations
+    val_pk = getattr(inst, field_meta.attname)
+    val_cls = field_meta.related_model
+    cache_key = _make_model_cache_key(val_cls, val_pk)
+    if cache_key in model_cache:
+        logger.debug("ev=model_cache, status=hit, key=%s", cache_key)
+        related_instance = model_cache[cache_key]
+    else:
+        logger.debug("ev=model_cache, status=miss, key=%s", cache_key)
+        related_instance = getattr(inst, field_name)
+        model_cache[cache_key] = related_instance
+    setattr(inst, field_name, related_instance)
```

Comment:
idea here is to pre-populate the related instance so if the function decides to access it or use it (normally it does) we don't do another query to the db. In example:

```python
FILTERS = {
    'counter': lambda inst: inst.counter.count + 1
}
```

When that function filter is called `counter` has already been populated by a counter from cache or stored for later used by another filter function.